src/control/mod.rs

```diff
@@ -336,12 +336,12 @@ impl<T: cio::CIO> Control<T> {
         }
     }
 
-    fn handle_lst_ev(&mut self, msg: Box<listener::Message>) {
+    fn handle_lst_ev(&mut self, msg: listener::Message) {
         debug!("Adding peer for torrent with hash {:?}!", msg.hash);
         if let Some(tid) = self.hash_idx.get(&msg.hash).cloned() {
             let id = msg.id;
             let rsv = msg.rsv;
-            self.add_inc_peer(tid, msg.peer, id, rsv);
+            self.add_inc_peer(tid, peer::PeerConn::new_incoming(msg.conn, msg.reader).unwrap(), id, rsv);
```

Comment:
Rather than unwrap can you have this do nothing if `new_incoming` fails.

Reply:
I copied the logic from the original location [here](https://github.com/Luminarys/synapse/pull/81/files#diff-6db71f1ee58ff4faea933522bc7610ceL92), which used `unwrap` when constructing the `PeerConn` on the other side of the channel. Should it not actually be infallible?

Reply:
There's a fair amount of places where IO calls which could fail in certain exceptional situations (mostly due to SIGINT) are unwrapped. I think at least in this case it's appropriate to handle the failure nicely.